src/jobs.rs

```diff
@@ -486,6 +505,156 @@ impl CommitJobInstance {
 
         Ok(json!({ "refs": commits }))
     }
+
+    fn use_storefront_info(
+        &self,
+        config: &Config,
+        build_ref: &BuildRef,
+        upload_path: &str,
+        conn: &PgConnection,
+    ) -> Result<Option<(StorefrontInfo, String)>, Box<dyn Error>> {
+        /* If applicable, get the storefront info for this app from the backend and apply it to the commit. This might
+        require rewriting the commit with a new dirtree. */
+
+        fn mtree_lookup(
+            mtree: &MutableTree,
+            path: &[&str],
+        ) -> Result<(Option<GString>, Option<MutableTree>), Box<dyn Error>> {
+            match path {
+                [file] => mtree.lookup(file).map_err(Into::into),
+                [subdir, rest @ ..] => mtree_lookup(
+                    &mtree
+                        .lookup(subdir)?
+                        .1
+                        .ok_or_else(|| "subdirectory not found".to_string())?,
+                    rest,
+                ),
+                [] => Err("no path given".into()),
+            }
+        }
+
+        let mtree_lookup_file = |mtree, path| -> Result<_, Box<dyn Error>> {
+            mtree_lookup(mtree, path)?
+                .0
+                .ok_or_else(|| "file not found".into())
+        };
+
+        let app_id = build_ref.ref_name.split('/').nth(1).unwrap();
+
+        if build_ref.ref_name.starts_with("app/") {
+            if let Some(endpoint) = &config.storefront_info_endpoint {
+                let convert_err =
+                    |e| format!("Failed to fetch storefront info from {}: {}", &endpoint, e);
+
+                /* Fetch the storefront info */
+                let storefront_info = reqwest::blocking::Client::new()
+                    .get(endpoint)
+                    .query(&[("app_id", app_id)])
+                    .send()
+                    .map_err(convert_err)?
+                    .error_for_status()?
+                    .json::<StorefrontInfo>()
+                    .map_err(convert_err)?;
+
+                /* Read the appstream file from the commit */
+                let repo = libostree::Repo::new_for_path(upload_path);
+                repo.open::<Cancellable>(None)?;
```

Comment:
You could spell this as

```suggestion
                repo.open(gio::NONE_CANCELLABLE)?;
```

which I think reads better.  (Here and elsewhere)